ale_python_interface/ale_c_wrapper.h

```diff
@@ -51,22 +50,40 @@ extern "C" {
   int getRAMSize(ALEInterface *ale){return ale->getRAM().size();}
   int getScreenWidth(ALEInterface *ale){return ale->getScreen().width();}
   int getScreenHeight(ALEInterface *ale){return ale->getScreen().height();}
+
   void getScreenRGB(ALEInterface *ale, unsigned char *screen_data){
-    int w = ale->getScreen().width();
-    int h = ale->getScreen().height();
-    int index = 0;
-    int r, g, b;
-    pixel_t *ale_screen_data = (pixel_t *)ale->getScreen().getArray();
-    for(int i = 0;i < w*h;i++){
-      r = (rgb_palette[ale_screen_data[i]] >> 16) & 0xFF;
-      g = (rgb_palette[ale_screen_data[i]] >>  8) & 0xFF;
-      b = (rgb_palette[ale_screen_data[i]] >>  0) & 0xFF;
-      screen_data[index++] = r;
-      screen_data[index++] = g;
-      screen_data[index++] = b;
+    size_t w = ale->getScreen().width();
+    size_t h = ale->getScreen().height();
+    size_t screen_size = w*h;
+
+    const uInt32 *current_palette = ale->theOSystem->colourPalette().getPalette();
+    pixel_t *ale_screen_data = ale->getScreen().getArray();
+    pixel_t *p = ale_screen_data;
+    pixel_t *q = screen_data;
 
+    for(size_t i = 0; i < screen_size; i++, p++){
+      int rgb = current_palette[*p];
+      *q = (unsigned char) ((rgb >> 16));  q++;    // r
+      *q = (unsigned char) ((rgb >>  8));  q++;    // g
+      *q = (unsigned char) ((rgb >>  0));  q++;    // b
     }
   }
+
+  void getScreenGrayscale(ALEInterface *ale, unsigned char *screen_data){
+    size_t w = ale->getScreen().width();
+    size_t h = ale->getScreen().height();
+    size_t screen_size = w*h;
+
+    const uInt32 *current_palette = ale->theOSystem->colourPalette().getPalette();
+    pixel_t *ale_screen_data = ale->getScreen().getArray();
+    pixel_t *p = ale_screen_data;
+    pixel_t *q = screen_data;
+
+    for(size_t i = 0; i < screen_size; i++, p++, q++){
+      *q = (unsigned char) (current_palette[*p+1] & 0xFF);
```

Comment:
If you think in those terms, I will move the whole thing to ColourPalette. I prefer to loop over pointers in the buffer than calling a method for each pixel. I know I could inline the method it but still. I can create some methods in ColourPalette that deal with buffers instead of single points. What do you think?